src/adapters/thinkerbell/mod.rs

```diff
@@ -0,0 +1,404 @@
+//! An adapter providing access to the Thinkerbell rules engine.
+
+use foxbox_taxonomy::adapter::{ AdapterManagerHandle, Adapter, WatchEvent, AdapterWatchGuard };
+use foxbox_taxonomy::api::{ API, Error, InternalError, ResultMap };
+use foxbox_taxonomy::services::{ Setter, Getter, AdapterId, ServiceId, Service, Channel, ChannelKind };
+use foxbox_taxonomy::util::Id;
+use foxbox_taxonomy::values::{ Range, Duration, Type, Value, TypeError, OnOff };
+use foxbox_thinkerbell::compile::ExecutableDevEnv;
+use foxbox_thinkerbell::manager::{ ScriptManager, ScriptId, Error as ScriptManagerError };
+use foxbox_thinkerbell::run::ExecutionEvent;
+
+use timer;
+use transformable_channels::mpsc::*;
+
+use std::collections::{ HashMap, HashSet };
+use std::path::Path;
+use std::sync::{ Arc, Mutex };
+use std::thread;
+
+static ADAPTER_NAME: &'static str = "Thinkerbell adapter (built-in)";
+static ADAPTER_VENDOR: &'static str = "team@link.mozilla.org";
+static ADAPTER_VERSION: [u32;4] = [0, 0, 0, 0];
+
+/// ThinkerbellAdapter hooks up the rules engine (if this, then that) as an adapter.
+///
+/// Each "rule", or "script", is a JSON-serialized structure according to Thinkerbell conventions.
+///
+/// This adapter exposes a root service, with one AddThinkerbellRule setter (to add a new rule).
```

Comment:
Note for the future: we'll probably want to add some kind of `PropertyBag` to getters/setters to convey information such as which `get enabled` matches with `set enabled`.